src/System.Management.Automation/engine/TypeMetadata.cs

```diff
@@ -1501,16 +1501,20 @@ private void RemoveParameter(CompiledCommandParameter parameter)
         ///
         private static bool IsMemberAParameter(MemberInfo member)
         {
-            bool result = false;
-
             try
             {
-                // MemberInfo.GetCustomAttributes returns IEnumerable<Attribute> in CoreCLR
-                var attributes = member.GetCustomAttributes(typeof(ParameterAttribute), false);
-                if (attributes.Any())
+                var expAttribute = member.GetCustomAttributes<ExperimentalAttribute>(false).FirstOrDefault();
+                if (expAttribute != null && expAttribute.ToHide) { return false; }
+
+                var hasAnyVisibleParamAttributes = false;
+                var paramAttributes = member.GetCustomAttributes<ParameterAttribute>(false);
+                foreach (var paramAttribute in paramAttributes)
                 {
-                    result = true;
+                    if (paramAttribute.ToHide) { continue; }
+                    hasAnyVisibleParamAttributes = true;
+                    break;
```

Comment:
There's a `break` followed immediately by a `return` here. Another way might be:

```csharp
foreach (var paramAttribute in member.GetCustomAttributes<ParameterAttribute>(inherit: false))
{
    if (paramAttribute.ToHide) { continue; }
    return true;
}
return false;
```

Reply:
Or possibly `return member.GetCustomAttributes<ParameterAttribute>(inherit: false).Any(paramAttribute => !paramAttribute.ToHide);`, but that's if LINQ isn't too heavyweight

Reply:
Maybe avoid LINQ as this is a hot code path. Changed the loop to the following:
```
foreach (var paramAttribute in paramAttributes)
{
    if (!paramAttribute.ToHide)
    {
        hasAnyVisibleParamAttributes = true;
        break;
    }
}
```

Reply:
Well my main point was that in terms of style that whole `hasAnyVisibleParamAttributes` variable only exists to return - there's no contingent logic on it. And when it gets set, we immediate break and return it. So it's not really needed as a variable at all.

Reply:
I personally think the variable name is useful for understanding the logic.